server/base/src/main/java/org/apache/accumulo/server/iterators/TabletIteratorEnvironment.java

```diff
@@ -1,269 +0,0 @@
-    this.fullMajorCompaction = false;
-    this.userCompaction = false;
-    this.authorizations = Authorizations.EMPTY;
-    this.topLevelIterators = new ArrayList<>();
-  }
-
-  public TabletIteratorEnvironment(ServerContext context, IteratorScope scope,
-      AccumuloConfiguration tableConfig, TableId tableId, SamplerConfigurationImpl samplerConfig) {
-    if (scope == IteratorScope.majc) {
-      throw new IllegalArgumentException("must set if compaction is full");
-    }
-
-    this.context = context;
-    this.serviceEnvironment = new ServiceEnvironmentImpl(context);
-    this.scope = scope;
-    this.trm = null;
-    this.tableConfig = tableConfig;
-    this.tableId = tableId;
-    this.fullMajorCompaction = false;
-    this.userCompaction = false;
-    this.authorizations = Authorizations.EMPTY;
-    if (samplerConfig != null) {
-      enableSampleForDeepCopy = true;
-      this.samplerConfig = samplerConfig.toSamplerConfiguration();
-    } else {
-      enableSampleForDeepCopy = false;
-    }
-    this.topLevelIterators = new ArrayList<>();
-  }
-
-  public TabletIteratorEnvironment(ServerContext context, IteratorScope scope,
-      AccumuloConfiguration tableConfig, TableId tableId, ScanFileManager trm,
-      Map<TabletFile,DataFileValue> files, Authorizations authorizations,
-      SamplerConfigurationImpl samplerConfig,
-      ArrayList<SortedKeyValueIterator<Key,Value>> topLevelIterators) {
-    if (scope == IteratorScope.majc) {
-      throw new IllegalArgumentException("must set if compaction is full");
-    }
-
-    this.context = context;
-    this.serviceEnvironment = new ServiceEnvironmentImpl(context);
-    this.scope = scope;
-    this.trm = trm;
-    this.tableConfig = tableConfig;
-    this.tableId = tableId;
-    this.fullMajorCompaction = false;
-    this.files = files;
-    this.authorizations = authorizations;
-    if (samplerConfig != null) {
-      enableSampleForDeepCopy = true;
-      this.samplerConfig = samplerConfig.toSamplerConfiguration();
-    } else {
-      enableSampleForDeepCopy = false;
-    }
-
-    this.topLevelIterators = topLevelIterators;
-  }
-
-  public TabletIteratorEnvironment(ServerContext context, IteratorScope scope, boolean fullMajC,
-      AccumuloConfiguration tableConfig, TableId tableId, CompactionKind kind) {
-    if (scope != IteratorScope.majc) {
-      throw new IllegalArgumentException(
-          "Tried to set maj compaction type when scope was " + scope);
-    }
-
-    this.context = context;
-    this.serviceEnvironment = new ServiceEnvironmentImpl(context);
-    this.scope = scope;
-    this.trm = null;
-    this.tableConfig = tableConfig;
-    this.tableId = tableId;
-    this.fullMajorCompaction = fullMajC;
-    this.userCompaction = kind.equals(CompactionKind.USER);
-    this.authorizations = Authorizations.EMPTY;
-    this.topLevelIterators = new ArrayList<>();
-  }
-
-  @Deprecated(since = "2.0.0")
-  @Override
-  public AccumuloConfiguration getConfig() {
```

Comment:
There will be a different configuration object impl returned for this methiod now that could have slightly different behavior.  Not sure if this would ever matter.

Reply:
From what I can tell, all of the places where this was used, it's based off `Context.getTableConfiguration(TableId)`.